updated policy 37 BWMPS to reflect new evidence shared by EAG

a-chacko opened the pull request:
I've updated the the script policy_37.R to reflect new weight loss and weight regain based on new papers suggested by EAG

All changes are only in this script [policy_37.R](https://github.com/nestauk/ahl_obesity_blueprint/blob/14-update-bwmp-inputs/analysis/policy_analysis/new_policies/policy_37.R)

Commit:
updated policy 37 BWMPS to reflect new evidence shared by EAG

Commit:
updates to BWMP (37) and FOP (35 policy scripts

Commit:
updates to policy 37, 38 along with comparison plots for 2 GLP-1 policies

a-chacko committed:
added .DS_Store to gitignore

a-chacko committed:
input changes

a-chacko committed:
tirzepatide sensitivity testing (#18)

* tirzepatide sensitivity testing

* sensitivity testing files and updated benefit calculation based on agreed method

* sensititvity tests with 5000 iterations

a-chacko committed:
updating scripts to align with latest assumptions

a-chacko committed:
modified policy 38 sensitivity testing script to report out class 3 obesity reduction

moniquetan commented on analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:
```diff
@@ -257,10 +257,27 @@ run_single_iteration <- function(seed_value,
+    mutate(freq = n/sum(n)*100) %>%
+    filter(bmi_class %in% c("morbidly obese")) %>%
+    summarise(baseline_class_3_obesity_prevalence = sum(freq))
+  
```
All good here. Essentially replicates the two blocks above, but just for morbid obesity, rather than obesity and morbid obesity.

a-chacko replied:
Yup, that's right!

moniquetan commented on analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:
```diff
@@ -257,10 +257,27 @@ run_single_iteration <- function(seed_value,
   
+  reduction_class_3 <- (class_3_baseline_prevalence$baseline_class_3_obesity_prevalence -
+    class_3_bmi_prevalence$class_3_obesity_prevalence_y5)
+  
```
OK too. Replicating what's above, but limited to class 3 obesity (instead of all obesity).

moniquetan commented on analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:
```diff
@@ -275,6 +292,10 @@ run_single_iteration <- function(seed_value,
+    year5_class_3_obesity = class_3_bmi_prevalence$class_3_obesity_prevalence_y5,
+    reduction_class_3 = reduction_class_3,
+    relative_reduction_class_3 = (reduction_class_3/class_3_baseline_prevalence$baseline_class_3_obesity_prevalence) * 100,
     benefit = value_to_gov,
```
OK here too

moniquetan commented on analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:
```diff
@@ -455,7 +476,11 @@ summary_stats <- sensitivity_results %>%
+    sd_class_3_rel_reduction = round(sd(relative_reduction_class_3), 3),
+    min_class_3_rel_reduction = round(min(relative_reduction_class_3), 3),
+    max_class_3_rel_reduction = round(max(relative_reduction_class_3), 3),
   )
```
all good here too

moniquetan reviewed (commented):
All good with amending the Tirzepatide sensitivity testing to get the class 3 obesity number. See point-by-point approvals in comments below.

I haven't reviewed the other changes made in earlier commits yet (eg BWMPs) and only focussed on the  Tirzepatide sensitivity testing for this comment/review.

I will do a separate review for BWMP.

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -55,6 +53,12 @@ source(file = "models/adult_model_calorie.R")
+NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS = 0.11
+BASELINE_FOPL_COVERAGE = 0.66
+COMPENSATION_EFFECT = 0.23
 
```
Constants match the slide here https://docs.google.com/presentation/d/1vW_9kjWoYoe_xvqEdC6rq15G9TT9F_fVnDik3OEQf98/edit?slide=id.g371f70717f1_0_1044#slide=id.g371f70717f1_0_1044, which is correct

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
-# Based on [A] and [C], the intake change = effect size - compensation effect = -9.57 kcals
+# calculating the change in intake:
+effect_size = 27
+effect_size_after_ndns_adjustment = effect_size * NDNS_ADJUSTMENT
```
correct, multiply by 1.32

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
+# calculating the change in intake:
+effect_size = 27
+effect_size_after_ndns_adjustment = effect_size * NDNS_ADJUSTMENT
+effect_size_ooh_adjustment = effect_size_after_ndns_adjustment * (1 - OOH_CONTRIBUTION_TO_DIETS)
```
correct, multiply by 1-0.11 =0.89

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
+effect_size = 27
+effect_size_after_ndns_adjustment = effect_size * NDNS_ADJUSTMENT
+effect_size_ooh_adjustment = effect_size_after_ndns_adjustment * (1 - OOH_CONTRIBUTION_TO_DIETS)
+effect_size_non_packaged_food_adjustment = effect_size_ooh_adjustment * (1 - NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS)
```
correct, multiply by 1-0.11=0.89

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
+effect_size_after_ndns_adjustment = effect_size * NDNS_ADJUSTMENT
+effect_size_ooh_adjustment = effect_size_after_ndns_adjustment * (1 - OOH_CONTRIBUTION_TO_DIETS)
+effect_size_non_packaged_food_adjustment = effect_size_ooh_adjustment * (1 - NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS)
+effect_size_baseline_coverage_adjustment = effect_size_non_packaged_food_adjustment * (1 - BASELINE_FOPL_COVERAGE)
```
correct, multiply by 1-0.66=0.34

moniquetan commented on analysis/policy_analysis/new_policies/policy_35.R:
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
+effect_size_ooh_adjustment = effect_size_after_ndns_adjustment * (1 - OOH_CONTRIBUTION_TO_DIETS)
+effect_size_non_packaged_food_adjustment = effect_size_ooh_adjustment * (1 - NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS)
+effect_size_baseline_coverage_adjustment = effect_size_non_packaged_food_adjustment * (1 - BASELINE_FOPL_COVERAGE)
+effect_size_post_compensation = effect_size_baseline_coverage_adjustment * (1 - COMPENSATION_EFFECT)
```
correct, multiply by 1-0.23=0.77

moniquetan reviewed (commented):
FOP review: reviewed the changes made since my last review (9 June), which consisted of adjusting the effect size around NDNS underestimation, OOH contribution, non-packaged food contribution, etc., as laid out in this slide https://docs.google.com/presentation/d/1vW_9kjWoYoe_xvqEdC6rq15G9TT9F_fVnDik3OEQf98/edit?slide=id.g371f70717f1_0_1044#slide=id.g371f70717f1_0_1044. 

I confirm that all adjustments have been applied correctly.

moniquetan commented on analysis/policy_analysis/new_policies/policy_36.R:
MUP: 
- The energy intake changes by alcohol consumption categories match those given in Table 4 here https://docs.google.com/document/d/1dWnjV9dU5eSV3GdUpn45eQzampI3e3iz4SRJCnmsGHs/edit?tab=t.0#heading=h.i0hc2zoax9, which is correct
- These different EI changes are then applied to different people in the population based on their alcohol intake. I assume that the levels in the `alcohol_overall` field have been used correctly.
- Line 117 is hardcoded, presumably due to `bmi_model` being a matrix rather than a df? The result may be correct but this handling is really risky and error-prone. 
   - Must do: double-check that the correct columns are selected via hard-coding
   - Should do: make necessary amends so that hardcoding his no longer necessary

a-chacko replied:
Hey Monique, Updates below:
**Must do:**
- I've checked and can confirm that the right columns are being chosen.

**Should do:**
- I'll add this as an issue on github to pick up later

a-chacko replied:
For the should do - #19 has been created to address this

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -10,37 +10,50 @@
 #   Allocated budget = £85 million per year from the policy specification
-# 
 #   Cost of delivering the programme: £70 per person
 #     Unit cost of treatment in 2015 = £51.45 per person [Source]
```
[Should] add missing source

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -10,37 +10,50 @@
-# 
 #   Cost of delivering the programme: £70 per person
 #     Unit cost of treatment in 2015 = £51.45 per person [Source]
 #     Adjusting for inflation = Original price x (CPI in 2015 / CPI in base year)
```
[Should] specify what the base year is + source of conversion factor

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -10,37 +10,50 @@
+#   people who enroll and the share of people who experience weight loss
+#   Number of people experiencing weight loss = 45% of those who enrolled
+#   And number of people experiencing weight loss of those who enrolled
+#           = 45% x 1,214,285 = 546428.6; rounding this to 546,429
```
[Should] this reads as if we are taking 45% of 45% of 1,2 million. But actually we are only applying the 45% once, not twice.

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
+
 # Constants
-NUMBER_OF_PEOPLE_PER_YEAR = 1200000
+NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS = number_enrolled_experiencing_weight_loss
```
nice to have it this way instead of hard-coding 1.2 million!

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
 # Constants
-NUMBER_OF_PEOPLE_PER_YEAR = 1200000
+NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS = number_enrolled_experiencing_weight_loss
 ENGLAND_ADULT_POPULATION = 44263393  # (a)
```
[should] (when revising codebase): take population figures out of policy-specific scripts and centralise

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -247,7 +275,7 @@ df = select_intervention_sample(data = df,
 
 # Assign weight changes to individuals who were selected in the previous step
-# We apply a weight loss of 2.4 kgs for those receiving the treatment and an annual weight regain of 0.32 kg
+# We apply a weight loss of 3.5 kgs for those receiving the treatment and an annual weight regain of 0.46 kg
```
[should] weight loss and weight regain values are wrong? doesn't affect the modelling as that's just comments, but can create confusion.

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -237,7 +265,7 @@ set.seed(370)
 # Selecting the intervention sample each year who will receive the intervention:
 df = select_intervention_sample(data = df,
-                                sample_size = NUMBER_OF_PEOPLE_PER_YEAR,
+                                sample_size = NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS,
```
correct to assign weight changes (weight loss + regain) only to those experiencing weight loss

moniquetan commented on analysis/policy_analysis/new_policies/policy_37.R:
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
+WEIGHT_REGAIN_POST_TREATMENT = 0.32 # from rapid review
 
 
 
```
Annoyingly can't comment on lines that were not modified as part of this PR, but the function `assign_weight_changes()` is correct in assigning weight regain only at year > 1 of treatment.  

[should] (in codebase revision): need to come up with naming convention to differentiate between year of modelling vs year of treatment.

a-chacko replied:
Created an issue [#23](https://github.com/nestauk/ahl_obesity_blueprint/issues/23) to address should comment

moniquetan reviewed (commented):
Hi @a-chacko ,

**BWMP impact code review**

A few 'should' comments to look into when revising the codebase. 

My only 'must' would be to give me the source of the cost pp of BWMP to ensure we treat the correct number of people? According to https://docs.google.com/spreadsheets/d/1MnA8i5HKPDkJbjHMyAyGtQKe7kYIhHPcyx9zQo8jSjM/edit?gid=829546672#gid=829546672 it would be the Taylor 2024 paper but I can't seem to find the £51.45 value in there.



> Hi @a-chacko ,
> 
> BWMP impact code review
> 
> A few 'should' comments to look into when revising the codebase.
> 
> My only 'must' would be to give me the source of the cost pp of BWMP to ensure we treat the correct number of people? According to https://docs.google.com/spreadsheets/d/1MnA8i5HKPDkJbjHMyAyGtQKe7kYIhHPcyx9zQo8jSjM/edit?gid=829546672#gid=829546672 it would be the Taylor 2024 paper but I can't seem to find the £51.45 value in there.
> 

The costs are now being calculated differently and the steps are captured on this slide - [BWMPs costs](https://docs.google.com/presentation/d/1tJZ3USk3l6S7kO6vB2uKHFZ_oyfXlRDO9REeCpx1M8s/edit?slide=id.g3811b0359be_0_1066#slide=id.g3811b0359be_0_1066)

cmbrennan002 commented on analysis/policy_analysis/new_policies/policy_38.R:
Question to discuss: Could we correct for this by rather that adding the weight to the 'weight_sum' column, we add 1/weight_sum, and then only stop drawing from that observation if the column reaches weight_sum. It would make the loop run a long time but might be feasible? If we think this is methodologically sound, can get some quick help to run this on EC2/ speed it up

a-chacko committed:
Option 2 sensitivity testing

a-chacko committed:
new sampling methodology (#21)

* new sampling methodology

* updating cohort variable

a-chacko committed:
updated the config file

a-chacko committed:
Four new policies (#13)

* testing

* scripts for adult and child model for Hall's dynamic weight modelling using HSE 2019 dataset

* initial commit

* Create README.md

* Update README.md

* Update README.md

* changes made to adult and child model codes

* commented out a line of code to install a package

* updated code with outputs

* updated changes to child pre processing

* updated output figues

* scripts for tranche 1 policies

* test

* 1. Added a new file with functions for the child model.
2. Updated the way of calculating bmi for children
3. Updated the method for creating the energy intake matrix
4. added 3 reference data files used for look up

* Scripts for policy 4a, 4b and 8b

* scripts for policies, 33, 4b, 5a and 5b

* added annotations to the script model_utils.R

* Implemented new functions to test a different approach for the child model

* Updated child modelling approaches and tranche 2 policies

* code for five policies from Tranches 2 and 3 - 13, 22, 20, 27 & 28

* re-arranging the files to a new folder - tranche 2b to include policies 13, 20, 22, 27 & 28

* Tranche 3 policies and changes to pre processing adults file with inclusion of new variables for selection

* policy 24 in tranche 3

* Updated the child modelling methodology for Tranche 1 policies

* Updated one of the source files to call in the scripts

* Added annotations to the child model script and functions

* Included annotations explaining functions called by child_model_calorie_henry.R from moel_utils.R

* tranche 4 policies and updates to other base scripts

* added new variables to read from health survey data

* Updates to policy analysis for different policies

* The following changes were made: folder changes

modelling updates

 removed sensitive info

* removed secrets ids

* updated gitignore

* latest modelling scripts

* commented scripts for analysis of variation in prevalence as a results of some policies in Scotland & Wales, compared to England

* updated the gitignore file to exclude outputs

* creating a default hall model script

* deleting output files

* updated the requirements file

* scripts with updated aggregated files (#12)

* scripts with updated aggregated files

* updated data creation file based on the comments in code review. Moved old scripts to archive and renamed the alt_* files with the original script names

---------

Co-authored-by: Anish Chacko <you@example.com>

* modelling of 4 new policies and associated files

* Remove output folder from tracking

* Add .DS_Store to gitignore

---------

Co-authored-by: Anish Chacko <you@example.com>

a-chacko committed:
Resolve merge conflicts with dev

a-chacko commented:
Both branches in this PR has already been merged into dev. Not deleting the branch for now, just in case we need it later on.
